app/services/safe_downloader.rb

```diff
@@ -20,22 +20,36 @@ def initialize(message = 'report is empty')
     RuntimeError,
     URI::InvalidURIError,
     DownloadError,
-    EmptyReportError
+    EmptyFileError
   ].freeze
 
   class << self
-    def download(url, max_size: nil)
-      downloaded_file = open_url(encode_url(url), create_options(max_size))
-      raise EmptyReportError if downloaded_file.size.zero?
+    def download(url, max_size: nil, ssl_only: false)
+      uri = encode_url(url, ssl_only)
+      downloaded_file = open_url(uri, create_options(max_size))
+      raise EmptyFileError if downloaded_file.size.zero?
 
-      report_contents(downloaded_file)
+      downloaded_file
     rescue *DOWNLOAD_ERRORS => e
       raise DownloadError if e.instance_of?(RuntimeError) &&
                              e.message !~ /redirection/
 
       raise DownloadError, "download failed (#{url}): #{e.message}"
     end
 
+    def download_reports(url, opts = {})
+      opts[:ssl_only] = Rails.env.production?
```

Comment:
Should this option be exposed higher up, maybe in the general app settings as `...__DOWNLOAD_SSL`?

Reply:
Refactored to be settable via `Settings.report_download_ssl_only`, that defaults to `true` for production. Thanks!